experimental/dds/tree2/src/feature-libraries/sequence-field/moveEffectTable.ts

```diff
@@ -108,35 +126,78 @@ export function isMoveMark<T>(mark: Mark<T>): mark is MoveMark<T> {
 	}
 }
 
-function applyMoveEffectsToDest<T>(
-	mark: MoveIn,
+export function isMoveDestination(effect: MarkEffect): effect is MoveDestination {
+	switch (effect.type) {
+		case "MoveIn":
+			return true;
+		default:
+			return false;
+	}
+}
+
+function adjustMoveEffectBasis<T>(effect: MoveEffectWithBasis<T>, newBasis: MoveId): MoveEffect<T> {
+	if (effect.basis === newBasis) {
+		return effect;
+	}
+
+	const adjusted = { ...effect, basis: newBasis };
+	const basisShift = newBasis - effect.basis;
+	if (effect.endpoint !== undefined) {
+		adjusted.endpoint = {
+			...effect.endpoint,
+			localId: brand(effect.endpoint.localId + basisShift),
+		};
+	}
+
+	// TODO: Handle splitting `movedMark`
```

Comment:
Why would this code need to worry about splitting `movedMark`?

Reply:
`movedMark` describes a range of cells. When shifting the basis (in practice we only increase it) we should discard the portion of `movedMark` which is describing cells which come before the start of the new basis.

Reply:
Would be good to either do it or update the comment with what you just wrote.